saw-remote-api/python/tests/saw/test_llvm_lax_pointer_ordering.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+import unittest
+from saw_client import *
+from saw_client.llvm import Contract, FreshVar, LLVMType, SetupVal, array_ty, cryptol, i64, void
+from saw_client.option import LaxPointerOrdering
+
+
+LEN = 42
+
+
+def ptr_to_fresh(c : Contract, ty : LLVMType, name : Optional[str] = None,
+                read_only : bool = False) -> Tuple[FreshVar, SetupVal]:
+    """Add to ``Contract`` ``c`` an allocation of a pointer of type ``ty`` initialized to an unknown fresh value.
+    If ``read_only == True`` then the allocated memory is immutable.
+
+    :returns A fresh variable bound to the pointers initial value and the newly allocated pointer. (The fresh
+             variable will be assigned ``name`` if provided/available.)"""
+    var = c.fresh_var(ty, name)
+    ptr = c.alloc(ty, points_to = var, read_only = read_only)
+    return (var, ptr)
+
+
+class ZipWithAddContract(Contract):
+    def specification(self):
+        array_t = array_ty(LEN, i64)
+        c_ptr = self.alloc(array_t)
+        (a, a_ptr) = ptr_to_fresh(self, name='a', ty=array_t, read_only=True)
+        (b, b_ptr) = ptr_to_fresh(self, name='b', ty=array_t, read_only=True)
+
+        self.execute_func(c_ptr, a_ptr, b_ptr)
+
+        self.points_to(c_ptr, cryptol(f'zipWith`{{ {LEN} }} (+) {a.name()} {b.name()}'))
+        self.returns(void)
+
+
+class LLVMPointerTest(unittest.TestCase):
+
+    def test_llvm_pointer(self):
+        connect(reset_server=True)
+        if __name__ == "__main__": view(LogResults())
+        bcname = str(Path('tests','saw','test-files', 'llvm_lax_pointer_ordering.bc'))
+        mod = llvm_load_module(bcname)
+
+        set_option(LaxPointerOrdering(), True)
```

Comment:
This is not the only possible API design. One could also imagine the following alternatives:

* `set_option(LaxPointerOrdering(true))`
* `enable_option(LaxPointerOrdering())` (and `disable_option(LaxPointerOrdering())`)

Let me know if you have a preference, especially if the design could be more Pythonic.

Reply:
I prefer the API you went with, but I wonder if we should go with the one in the second bullet instead because it's closer to the syntax used in SAW scripts.

Reply:
We certainly aren't required to mimic the design of SAWScript, and we've deviated from how SAWScript does things already in various places (e.g., having a single `points_to` function instead of SAWScript's multitude of different `llvm_points_to*` functions). Since you prefer the API here, I think I'll just use that. In fact, now that I look closer, there's already somewhat of a convention established for passing `bool`s for configuration purposes in the [`logging`](https://github.com/GaloisInc/saw-script/blob/0b040994edce982f3bd882277242b980c87f8637/saw-remote-api/python/saw_client/connection.py#L117) function.